packages/tupaia-web/src/api/api.ts

```diff
@@ -57,8 +60,28 @@ const request = async (endpoint: string, options?: RequestParametersWithMethod)
   }
 };
 
-export const get = (endpoint: string, options?: RequestParameters) =>
-  request(endpoint, { method: 'get', ...options });
+/**
+ * This method handles query cancellation. Because we are using an older version of react-query, we still need to handle cancellation the old way: https://tanstack.com/query/v3/docs/react/guides/query-cancellation#old-cancel-function
+ */
+export const get = (endpoint: string, options?: RequestParameters) => {
+  // cancelOnUnmount is true by default. It can be opted out of by setting cancelOnUnmount to false in the options object
+  const { cancelOnUnmount = true, ...rest } = options || {};
+
+  const controller = cancelOnUnmount ? new AbortController() : null;
+  const promise = request(endpoint, {
+    method: 'get',
+    ...rest,
+    signal: controller?.signal,
+  }) as Promise<any> & {
+    cancel?: () => void;
+  };
+
+  if (cancelOnUnmount) {
+    promise.cancel = () => controller?.abort();
+  }
```

Comment:
For some reason, even though we are on v3.39 and cancellation when you pass in a signal was included form v3, I couldn't get cancellation automatically working unless I handled it using this old way. So possibly it is an issue that was fixed later on